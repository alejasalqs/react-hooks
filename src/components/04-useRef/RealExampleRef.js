import React, { useState } from "react";
import "../02-useEffect/effects.css";
import { MultipleCustomHooks } from "../examples/MultipleCustomHooks";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real Example Ref</h1>
      <hr />

      {show && <MultipleCustomHooks />}
      <br />
      <button className="btn btn-primary mt-5" onClick={() => setShow(!show)}>
        Show quotes
      </button>
    </div>
  );
};
