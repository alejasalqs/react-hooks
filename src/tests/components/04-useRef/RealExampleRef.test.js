import React from "react";
import Adapter from "enzyme-adapter-react-16";
const { shallow, configure } = require("enzyme");
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

configure({ adapter: new Adapter() });

describe("Pruebas en el archivo <RealExampleRef />", () => {
  const wrapper = shallow(<RealExampleRef />);
  test("Debe de mostrarse correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
    //expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("Debe ocultar componente <MultipleCustomHooks /> al dar click en el botÃ³n ", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
