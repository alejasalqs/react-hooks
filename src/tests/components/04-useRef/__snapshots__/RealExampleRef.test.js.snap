// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <RealExampleRef /> Debe de mostrarse correctamente  1`] = `ShallowWrapper {}`;
